Backported from 5.6.25 by Remi.

From f973877a2f8d58b857f0f02b8a88a2ee05a1cbb0 Mon Sep 17 00:00:00 2001
From: Stanislav Malyshev <stas@php.net>
Date: Sun, 14 Aug 2016 23:13:30 -0700
Subject: [PATCH] Fix bug #72836 - integer overflow in base64_decode caused
 heap corruption

From f01446dacf3eeab888b500115f0d71df7918c353 Mon Sep 17 00:00:00 2001
From: Stanislav Malyshev <stas@php.net>
Date: Tue, 16 Aug 2016 16:34:35 -0700
Subject: [PATCH] Fix TSRM build

Source: Remi Collet scl-php55, https://git.remirepo.net/cgit/rpms/scl-php55/php.git/tree/bug72836.patch
---
diff --git a/ext/standard/base64.c b/ext/standard/base64.c
--- a/ext/standard/base64.c
+++ b/ext/standard/base64.c
@@ -66,6 +66,12 @@ PHPAPI unsigned char *php_base64_encode(const unsigned char *str, int length, in
 		return NULL;
 	}
 
+	if (((size_t)length + 2) / 3 > INT_MAX/4 ) {
+		TSRMLS_FETCH();
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "String too long, maximum is %d", INT_MAX/4);
+		return NULL;
+	}
+
 	result = (unsigned char *) safe_emalloc((length + 2) / 3, 4 * sizeof(char), 1);
 	p = result;
 
